src/libraries/System.Private.CoreLib/src/System/Globalization/OrdinalCasing.Icu.cs

```diff
@@ -85,7 +92,12 @@ internal static char ToUpper(char c)
                 return (char) s_basicLatin[(int)c];
             }
 
-            ushort[]? casingTable = s_casingTable[pageNumber];
+            if (s_casingTable == null)
```

Comment:
would be better to remove this check and initialize s_casingTable statically. something like 

static ... s_casingTable =  InitCasingTable();

or something similar. That will avoid this extra check.